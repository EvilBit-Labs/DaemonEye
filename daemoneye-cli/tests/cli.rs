use assert_cmd::prelude::*;
use insta::assert_snapshot;
use std::process::Command;

#[test]
fn prints_expected_greeting() -> Result<(), Box<dyn std::error::Error>> {
    use tempfile::tempdir;

    let temp_dir = tempdir()?;
    let db_path = temp_dir.path().join("test.db");

    let mut cmd = Command::cargo_bin("daemoneye-cli")?;
    cmd.arg("--database").arg(&db_path);

    let output = cmd.output()?;
    if !output.status.success() {
        let stderr = String::from_utf8_lossy(&output.stderr);
        eprintln!("Command failed with stderr: {}", stderr);
    }
    assert!(output.status.success());
    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!("daemoneye-cli_greeting", stdout);
    Ok(())
}
