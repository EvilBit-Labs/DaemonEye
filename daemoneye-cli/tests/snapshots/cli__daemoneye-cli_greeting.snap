---
source: daemoneye-cli/tests/cli.rs
expression: stdout
---
DaemonEye Database Statistics
============================
Processes: 0
Rules: 0
Alerts: 0
System Info: 0
Scans: 0
Health status: healthy
daemoneye-cli completed successfully
